services/movie_session.py

```diff
@@ -0,0 +1,80 @@
+from datetime import datetime
+from typing import List
+
+from django.utils.dateparse import parse_date
+
+from db.models import CinemaHall, Movie, MovieSession
+
+
+def create_movie_session(movie_show_time: datetime,
+                         movie_id: int,
+                         cinema_hall_id: int) -> None:
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id,
+    )
+
+
+def get_movies_sessions(session_date: datetime = None) -> List[MovieSession]:
+    """
+    Retrieve movie sessions, optionally filtered by a specific date.
+    """
+    # If a session date is provided, parse it and filter sessions
+    if session_date:
+        date = parse_date(session_date)
+        if date:
+            return MovieSession.objects.filter(show_time__date=date)
+
+    # If no session date is provided, return all movie sessions
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
```

Comment:
The function `get_movie_session_by_id` is annotated to return a `MovieSession`, but it returns `None` if the session does not exist. Consider updating the return type annotation to `Optional[MovieSession]` to accurately reflect the possible return values.